server/src/types.rs

```diff
@@ -220,7 +228,26 @@ mod tests {
     use test_case::test_case;
     use tracing::warn;
 
-    use crate::types::EdgeToken;
+    use crate::types::{EdgeToken, TokenType};
+
+    fn test_str(token: &str) -> EdgeToken {
+        EdgeToken::from_str(
+            &(token.to_owned() + ".614a75cf68bef8703aa1bd8304938a81ec871f86ea40c975468eabd6"),
+        )
+        .unwrap()
+    }
+
+    fn test_token(env: Option<&str>, projects: Vec<&str>) -> EdgeToken {
```

Comment:
So typically what I would do here is derive default on the underlying struct, and then you can use destructuring to populate the fields you want:

```
EdgeToken{
  token_type: Some(TokenType::Client),
  ..EdgeToken::default()
}
```